ouroboros-consensus/src/Ouroboros/Storage/ChainDB/Impl/Iterator.hs

```diff
@@ -452,8 +456,8 @@ data IteratorState m blk
     --
     -- Invariant: the iterator is not exhausted.
   | InVolDB
-      (StreamFrom blk)
-      (NonEmpty (HeaderHash blk))
+      !(StreamFrom blk)
+      !(NonEmpty (HeaderHash blk))
```

Comment:
I need to investigate why this isn't giving us thunks (see other constructors of this type).